pallets/pallet-did-lookup/src/linkable_account.rs

```diff
@@ -0,0 +1,110 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::RuntimeDebug;
+use scale_info::TypeInfo;
+use sp_runtime::{AccountId32, MultiSignature, MultiSigner};
+
+use crate::{
+	account::{AccountId20, EthereumSignature, EthereumSigner},
+	signature,
+};
+
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub enum LinkableAccountId {
+	AccountId20(AccountId20),
+	AccountId32(AccountId32),
+}
+
+impl From<AccountId20> for LinkableAccountId {
+	fn from(account_id: AccountId20) -> Self {
+		Self::AccountId20(account_id)
+	}
+}
+
+impl From<AccountId32> for LinkableAccountId {
+	fn from(account_id: AccountId32) -> Self {
+		Self::AccountId32(account_id)
+	}
+}
+
+impl signature::GetWrapType for LinkableAccountId {
+	fn get_wrap_type(&self) -> signature::WrapType {
+		match self {
+			Self::AccountId20(_) => signature::WrapType::Dotsama,
+			Self::AccountId32(_) => signature::WrapType::Ethereum,
```

Comment:
Incorrect matching 😉 I would also rename Dotsama to Substrate
```suggestion
			Self::AccountId20(_) => signature::WrapType::Ethereum,
			Self::AccountId32(_) => signature::WrapType::Dotsama,
```

Reply:
Thanks :) fixed it.